### DISCUSSION TITLE
How to configure memory limit for Next.js API functions

### DISCUSSION THREAD

**Discussion by bravetheheat** [vercel/community#183] [Idea]
I have a number of Serverless functions that require much less than the default 1024MB of RAM while running (simple proxies). They are configured with my Next.js application. However, I cannot find any documentation of how to _reduce_ (or increase) this memory limit.

I have tried configuring directly from vercel.json by referencing the full relative path of the API function, but this opts me out of the "@vercel/next" optimized lambdas mode.

Is there a way to lower the memory allocation _without_ opting out of optimizations? Thanks.

**Comment by meotimdihia** [comment]
Hi, @bravetheheat. I think this link will solve your problem.
https://vercel.com/docs/configuration?query=vercel%20#project/functions

**Reply by bravetheheat** [reply]
@meotimdihia, thanks for your response. I mentioned that I'm looking for a way to configure the limit without touching the vercel.json configuration, which appears to opt-out my program from optimized lambdas mode. 

Maybe there's no way and the problem is more with the mode switch even with changes to just the API functions, but if so it would be great to have it confirmed.

**Comment by remorses** [comment]
Did anyone find any solution?

**Comment by thiagoenge** [comment]
I face up same problem, i didnt find a solution yet.

**Comment by mcfedr** [comment]
I've been trying to do this with `vercel.json`, but cannot seem to get it to work, whatever I put there 

something like this

```json
{
  "functions": {
    "**": {
      "memory": 128
    }
  }
}
```


```bash
$ npx vercel build
Vercel CLI 24.2.5 build (beta) — https://vercel.com/feedback
Error! The pattern "*" defined in `functions` doesn't match any Serverless Functions.
```

but doesnt want to work

**Comment by mcfedr** [comment]
5 minutes later, i worked it out, in `vercel.json`, the error message was leading me there, https://vercel.com/docs/errors#errors/unmatched-function-pattern

```json
{
  "functions": {
    "pages/**/*": {
      "memory": 128
    }
  }
}
```

**Reply by leerob** [reply]
Out of curiosity, why are you looking to lower memory?

**Reply by mcfedr** [reply]
Because i want to reduce the GB-Hours that I'm using on Vercel

> The value is given in GB-Hours, which is the memory allocated for each Function in GB, multiplied by the time in hours they were running

so if i change the allocation from 1GB to 128MB, I'm going to be using 8x less.

**Reply by Korayem** [reply]
@leerob it's not clear in the pricing docs whether vercel automatically sets proper memory or we have to think about that to optimize costs as shown by @mcfedr 

What if vercel default of 1024MB is too much assigned per function, do we get billed by assigned memory or actual consumed memory?
